src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_CallerInfoAttributes.cs

```diff
@@ -17,6 +17,400 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
 {
     public class AttributeTests_CallerInfoAttributes : WellKnownAttributesTestBase
     {
+        [Fact]
+        public void TestGoodCallerArgumentExpressionAttribute()
+        {
+            string source = @"
+using System;
+using System.Runtime.CompilerServices;
+
+class Program
+{
+    public static void Main()
+    {
+        Log(123);
+    }
+    const string p = nameof(p);
+    static void Log(int p, [CallerArgumentExpression(p)] string arg = null)
+    {
+        Console.WriteLine(arg);
+    }
+}
+";
+
+            var compilation = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.ReleaseExe, parseOptions: CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.Preview));
```

Comment:
>CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.Preview) [](start = 148, length = 71)

TestOptions.RegularPreview?

Reply:
Will update.

Note: tests (currently) only pass on .NET 5.0 test runs. Have no idea how to fix them (https://github.com/dotnet/roslyn/pull/51952#issuecomment-803289523).

Reply:
>  tests (currently) only pass on .NET 5.0 test runs. Have no idea how to fix them

Does the test output give any hint what it doesn't like?

---
In reply to: [601895861](https://github.com/dotnet/roslyn/pull/51952#discussion_r601895861) [](ancestors = 601895861)

Reply:
I think it is likely due to the fact that you unconditionally using TargetFramework.NetCoreApp

---
In reply to: [601905765](https://github.com/dotnet/roslyn/pull/51952#discussion_r601905765) [](ancestors = 601905765,601895861)

Reply:
It looks like the attribute is available only on NetCore. If that is the case and the tests fail on desktop because of missing type when TargetFramework.Standard/TargetFramework.StandardLatest is used, then we might stick with TargetFramework.NetCoreApp and skipping verification when run on desktop (RuntimeUtilities.IsDesktopRuntime). However, it would be good to see the actual PEVerify error.

---
In reply to: [601906492](https://github.com/dotnet/roslyn/pull/51952#discussion_r601906492) [](ancestors = 601906492,601905765,601895861)

Reply:
@AlekseyTs The tests currently already skips verification, but still fails *only* on net472 test runs.

The failure message with skipped verification is:

<details>
<summary>Click</summary>

```
   System.IO.FileLoadException : Could not load file or assembly 'System.Runtime, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. General Exception (Exception from HRESULT: 0x80131500)
---- System.Exception : Exception loading System.Runtime reflectionOnly:False
-------- System.BadImageFormatException : Cannot load a reference assembly for execution.
  Stack Trace:
  
Server stack trace: 
   at System.Reflection.RuntimeAssembly.GetEntryPoint(RuntimeAssembly assembly, ObjectHandleOnStack retMethod)
   at System.Reflection.RuntimeAssembly.get_EntryPoint()
   at Roslyn.Test.Utilities.Desktop.RuntimeAssemblyManager.Execute(String moduleName, String[] mainArgs, Nullable`1 expectedOutputLength, String& output) in /_/src/Compilers/Test/Core/Platform/Desktop/RuntimeAssemblyManager.cs:line 390
   at System.Runtime.Remoting.Messaging.StackBuilderSink._PrivateProcessMessage(IntPtr md, Object[] args, Object server, Object[]& outArgs)
   at System.Runtime.Remoting.Messaging.StackBuilderSink.SyncProcessMessage(IMessage msg)

Exception rethrown at [0]: 
   at System.Runtime.Remoting.Proxies.RealProxy.HandleReturnMessage(IMessage reqMsg, IMessage retMsg)
   at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke(MessageData& msgData, Int32 type)
   at Roslyn.Test.Utilities.Desktop.RuntimeAssemblyManager.Execute(String moduleName, String[] mainArgs, Nullable`1 expectedOutputLength, String& output)
   at Roslyn.Test.Utilities.Desktop.DesktopRuntimeEnvironment.Execute(String moduleName, String[] args, String expectedOutput) in /_/src/Compilers/Test/Core/Platform/Desktop/DesktopRuntimeEnvironment.cs:line 244
   at Microsoft.CodeAnalysis.Test.Utilities.CompilationVerifier.Emit(String expectedOutput, Nullable`1 expectedReturnCode, String[] args, IEnumerable`1 manifestResources, EmitOptions emitOptions, Verification peVerify, SignatureDescription[] expectedSignatures) in /_/src/Compilers/Test/Core/CompilationVerifier.cs:line 214
   at Microsoft.CodeAnalysis.Test.Utilities.CommonTestBase.Emit(Compilation compilation, IEnumerable`1 dependencies, IEnumerable`1 manifestResources, SignatureDescription[] expectedSignatures, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, Action`1 assemblyValidator, Action`1 symbolValidator, EmitOptions emitOptions, Verification verify) in /_/src/Compilers/Test/Core/CommonTestBase.cs:line 153
   at Microsoft.CodeAnalysis.Test.Utilities.CommonTestBase.CompileAndVerifyCommon(Compilation compilation, IEnumerable`1 manifestResources, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 assemblyValidator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, EmitOptions emitOptions, Verification verify) in /_/src/Compilers/Test/Core/CommonTestBase.cs:line 67
   at Microsoft.CodeAnalysis.CSharp.Test.Utilities.CSharpTestBase.CompileAndVerify(Compilation compilation, IEnumerable`1 manifestResources, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 validator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, EmitOptions emitOptions, Verification verify) in /_/src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs:line 830
   at Microsoft.CodeAnalysis.CSharp.UnitTests.AttributeTests_CallerInfoAttributes.TestCallerArgumentExpressionWithMultipleOptionalAttribute() in /_/src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_CallerInfoAttributes.cs:line 312
----- Inner Stack Trace -----
   at Roslyn.Test.Utilities.Desktop.DesktopRuntimeUtil.LoadAsAssembly(String moduleName, ImmutableArray`1 rawAssembly, Boolean reflectionOnly) in /_/src/Compilers/Test/Core/Platform/Desktop/DesktopRuntimeUtil.cs:line 98
   at Roslyn.Test.Utilities.Desktop.AppDomainAssemblyCache.GetOrLoad(ModuleData moduleData, Boolean reflectionOnly) in /_/src/Compilers/Test/Core/Platform/Desktop/AppDomainAssemblyCache.cs:line 81
   at Roslyn.Test.Utilities.Desktop.RuntimeAssemblyManager.GetAssembly(String fullName, Boolean reflectionOnly) in /_/src/Compilers/Test/Core/Platform/Desktop/RuntimeAssemblyManager.cs:line 322
   at Roslyn.Test.Utilities.Desktop.RuntimeAssemblyManager.AssemblyResolve(ResolveEventArgs args, Boolean reflectionOnly) in /_/src/Compilers/Test/Core/Platform/Desktop/RuntimeAssemblyManager.cs:line 298
   at Roslyn.Test.Utilities.Desktop.RuntimeAssemblyManager.AssemblyResolve(Object sender, ResolveEventArgs args) in /_/src/Compilers/Test/Core/Platform/Desktop/RuntimeAssemblyManager.cs:line 285
   at System.AppDomain.OnAssemblyResolveEvent(RuntimeAssembly assembly, String assemblyFullName)
----- Inner Stack Trace -----
   at System.Reflection.RuntimeAssembly.nLoadImage(Byte[] rawAssembly, Byte[] rawSymbolStore, Evidence evidence, StackCrawlMark& stackMark, Boolean fIntrospection, Boolean fSkipIntegrityCheck, SecurityContextSource securityContextSource)
   at System.Reflection.Assembly.Load(Byte[] rawAssembly)
   at Roslyn.Test.Utilities.Desktop.DesktopRuntimeUtil.LoadAsAssembly(String moduleName, ImmutableArray`1 rawAssembly, Boolean reflectionOnly) in /_/src/Compilers/Test/Core/Platform/Desktop/DesktopRuntimeUtil.cs:line 93
```

</details>

A different message when pe verification is also skipped:

<details>
<summary>Click</summary>

```
  Failed Microsoft.CodeAnalysis.CSharp.UnitTests.AttributeTests_CallerInfoAttributes.TestGoodCallerArgumentExpressionAttribute_DifferentAssembly [22 ms]
  Error Message:
   System.Exception : Exception loading System.Runtime reflectionOnly:False
---- System.BadImageFormatException : Cannot load a reference assembly for execution.
  Stack Trace:
  
Server stack trace: 
   at Roslyn.Test.Utilities.Desktop.DesktopRuntimeUtil.LoadAsAssembly(String moduleName, ImmutableArray`1 rawAssembly, Boolean reflectionOnly) in /_/src/Compilers/Test/Core/Platform/Desktop/DesktopRuntimeUtil.cs:line 98
   at Roslyn.Test.Utilities.Desktop.AppDomainAssemblyCache.GetOrLoad(ModuleData moduleData, Boolean reflectionOnly) in /_/src/Compilers/Test/Core/Platform/Desktop/AppDomainAssemblyCache.cs:line 81
   at Roslyn.Test.Utilities.Desktop.RuntimeAssemblyManager.GetAssembly(String fullName, Boolean reflectionOnly) in /_/src/Compilers/Test/Core/Platform/Desktop/RuntimeAssemblyManager.cs:line 322
   at Roslyn.Test.Utilities.Desktop.RuntimeAssemblyManager.AssemblyResolve(ResolveEventArgs args, Boolean reflectionOnly) in /_/src/Compilers/Test/Core/Platform/Desktop/RuntimeAssemblyManager.cs:line 298
   at Roslyn.Test.Utilities.Desktop.RuntimeAssemblyManager.AssemblyResolve(Object sender, ResolveEventArgs args) in /_/src/Compilers/Test/Core/Platform/Desktop/RuntimeAssemblyManager.cs:line 285
   at System.AppDomain.OnAssemblyResolveEvent(RuntimeAssembly assembly, String assemblyFullName)
   at System.Reflection.RuntimeAssembly.GetEntryPoint(RuntimeAssembly assembly, ObjectHandleOnStack retMethod)
   at System.Reflection.RuntimeAssembly.get_EntryPoint()
   at Roslyn.Test.Utilities.Desktop.RuntimeAssemblyManager.Execute(String moduleName, String[] mainArgs, Nullable`1 expectedOutputLength, String& output) in /_/src/Compilers/Test/Core/Platform/Desktop/RuntimeAssemblyManager.cs:line 390
   at System.Runtime.Remoting.Messaging.StackBuilderSink._PrivateProcessMessage(IntPtr md, Object[] args, Object server, Object[]& outArgs)
   at System.Runtime.Remoting.Messaging.StackBuilderSink.SyncProcessMessage(IMessage msg)

Exception rethrown at [0]: 
   at System.Runtime.Remoting.Proxies.RealProxy.HandleReturnMessage(IMessage reqMsg, IMessage retMsg)
   at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke(MessageData& msgData, Int32 type)
   at Roslyn.Test.Utilities.Desktop.RuntimeAssemblyManager.Execute(String moduleName, String[] mainArgs, Nullable`1 expectedOutputLength, String& output)
   at Roslyn.Test.Utilities.Desktop.DesktopRuntimeEnvironment.Execute(String moduleName, String[] args, String expectedOutput) in /_/src/Compilers/Test/Core/Platform/Desktop/DesktopRuntimeEnvironment.cs:line 244
   at Microsoft.CodeAnalysis.Test.Utilities.CompilationVerifier.Emit(String expectedOutput, Nullable`1 expectedReturnCode, String[] args, IEnumerable`1 manifestResources, EmitOptions emitOptions, Verification peVerify, SignatureDescription[] expectedSignatures) in /_/src/Compilers/Test/Core/CompilationVerifier.cs:line 214
   at Microsoft.CodeAnalysis.Test.Utilities.CommonTestBase.Emit(Compilation compilation, IEnumerable`1 dependencies, IEnumerable`1 manifestResources, SignatureDescription[] expectedSignatures, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, Action`1 assemblyValidator, Action`1 symbolValidator, EmitOptions emitOptions, Verification verify) in /_/src/Compilers/Test/Core/CommonTestBase.cs:line 153
   at Microsoft.CodeAnalysis.Test.Utilities.CommonTestBase.CompileAndVerifyCommon(Compilation compilation, IEnumerable`1 manifestResources, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 assemblyValidator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, EmitOptions emitOptions, Verification verify) in /_/src/Compilers/Test/Core/CommonTestBase.cs:line 67
   at Microsoft.CodeAnalysis.CSharp.Test.Utilities.CSharpTestBase.CompileAndVerify(Compilation compilation, IEnumerable`1 manifestResources, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 validator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, EmitOptions emitOptions, Verification verify) in /_/src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs:line 830
   at Microsoft.CodeAnalysis.CSharp.UnitTests.AttributeTests_CallerInfoAttributes.TestGoodCallerArgumentExpressionAttribute_DifferentAssembly() in /_/src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_CallerInfoAttributes.cs:line 133
----- Inner Stack Trace -----
   at System.Reflection.RuntimeAssembly.nLoadImage(Byte[] rawAssembly, Byte[] rawSymbolStore, Evidence evidence, StackCrawlMark& stackMark, Boolean fIntrospection, Boolean fSkipIntegrityCheck, SecurityContextSource securityContextSource)
   at System.Reflection.Assembly.Load(Byte[] rawAssembly)
   at Roslyn.Test.Utilities.Desktop.DesktopRuntimeUtil.LoadAsAssembly(String moduleName, ImmutableArray`1 rawAssembly, Boolean reflectionOnly) in /_/src/Compilers/Test/Core/Platform/Desktop/DesktopRuntimeUtil.cs:line 93
```

</details>

With un-skipped verification:

<details>
<summary>Click</summary>

```
  Failed Microsoft.CodeAnalysis.CSharp.UnitTests.AttributeTests_CallerInfoAttributes.TestArgumentExpressionIsCallerMember [18 ms]
  Error Message:
   System.Exception : Verification failed
---- Roslyn.Test.Utilities.Desktop.RuntimePeVerifyException : 
PeVerify failed for assembly '':
Type load failed.
Type load failed.


  Stack Trace:
     at Roslyn.Test.Utilities.Desktop.DesktopRuntimeEnvironment.Verify(Verification verification) in /_/src/Compilers/Test/Core/Platform/Desktop/DesktopRuntimeEnvironment.cs:line 324
   at Microsoft.CodeAnalysis.Test.Utilities.CompilationVerifier.Emit(String expectedOutput, Nullable`1 expectedReturnCode, String[] args, IEnumerable`1 manifestResources, EmitOptions emitOptions, Verification peVerify, SignatureDescription[] expectedSignatures) in /_/src/Compilers/Test/Core/CompilationVerifier.cs:line 205
   at Microsoft.CodeAnalysis.Test.Utilities.CommonTestBase.Emit(Compilation compilation, IEnumerable`1 dependencies, IEnumerable`1 manifestResources, SignatureDescription[] expectedSignatures, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, Action`1 assemblyValidator, Action`1 symbolValidator, EmitOptions emitOptions, Verification verify) in /_/src/Compilers/Test/Core/CommonTestBase.cs:line 153
   at Microsoft.CodeAnalysis.Test.Utilities.CommonTestBase.CompileAndVerifyCommon(Compilation compilation, IEnumerable`1 manifestResources, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 assemblyValidator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, EmitOptions emitOptions, Verification verify) in /_/src/Compilers/Test/Core/CommonTestBase.cs:line 67
   at Microsoft.CodeAnalysis.CSharp.Test.Utilities.CSharpTestBase.CompileAndVerify(Compilation compilation, IEnumerable`1 manifestResources, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 validator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, EmitOptions emitOptions, Verification verify) in /_/src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs:line 830
   at Microsoft.CodeAnalysis.CSharp.UnitTests.AttributeTests_CallerInfoAttributes.TestArgumentExpressionIsCallerMember() in /_/src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_CallerInfoAttributes.cs:line 385
----- Inner Stack Trace -----
   at Roslyn.Test.Utilities.Desktop.RuntimeAssemblyManager.PeVerifyModules(String[] modulesToVerify, Boolean throwOnError) in /_/src/Compilers/Test/Core/Platform/Desktop/RuntimeAssemblyManager.cs:line 459
   at Roslyn.Test.Utilities.Desktop.RuntimeAssemblyManager.PeVerifyModules(String[] modulesToVerify, Boolean throwOnError)
   at Roslyn.Test.Utilities.Desktop.DesktopRuntimeEnvironment.Verify(Verification verification) in /_/src/Compilers/Test/Core/Platform/Desktop/DesktopRuntimeEnvironment.cs:line 314
[xUnit.net 00:00:19.04]     Microsoft.CodeAnalysis.CSharp.UnitTests.AttributeTests_CallerInfoAttributes.TestGoodCallerArgumentExpressionAttribute_SwapArguments [FAIL]
  Failed Microsoft.CodeAnalysis.CSharp.UnitTests.AttributeTests_CallerInfoAttributes.TestGoodCallerArgumentExpressionAttribute_SwapArguments [24 ms]
  Error Message:
   System.Exception : Exception loading System.Runtime reflectionOnly:False
---- System.BadImageFormatException : Cannot load a reference assembly for execution.
  Stack Trace:
  
Server stack trace: 
   at Roslyn.Test.Utilities.Desktop.DesktopRuntimeUtil.LoadAsAssembly(String moduleName, ImmutableArray`1 rawAssembly, Boolean reflectionOnly) in /_/src/Compilers/Test/Core/Platform/Desktop/DesktopRuntimeUtil.cs:line 98
   at Roslyn.Test.Utilities.Desktop.AppDomainAssemblyCache.GetOrLoad(ModuleData moduleData, Boolean reflectionOnly) in /_/src/Compilers/Test/Core/Platform/Desktop/AppDomainAssemblyCache.cs:line 81
   at Roslyn.Test.Utilities.Desktop.RuntimeAssemblyManager.GetAssembly(String fullName, Boolean reflectionOnly) in /_/src/Compilers/Test/Core/Platform/Desktop/RuntimeAssemblyManager.cs:line 322
   at Roslyn.Test.Utilities.Desktop.RuntimeAssemblyManager.AssemblyResolve(ResolveEventArgs args, Boolean reflectionOnly) in /_/src/Compilers/Test/Core/Platform/Desktop/RuntimeAssemblyManager.cs:line 298
   at Roslyn.Test.Utilities.Desktop.RuntimeAssemblyManager.AssemblyResolve(Object sender, ResolveEventArgs args) in /_/src/Compilers/Test/Core/Platform/Desktop/RuntimeAssemblyManager.cs:line 285
   at System.AppDomain.OnAssemblyResolveEvent(RuntimeAssembly assembly, String assemblyFullName)
   at System.Reflection.RuntimeAssembly.GetEntryPoint(RuntimeAssembly assembly, ObjectHandleOnStack retMethod)
   at System.Reflection.RuntimeAssembly.get_EntryPoint()
   at Roslyn.Test.Utilities.Desktop.RuntimeAssemblyManager.Execute(String moduleName, String[] mainArgs, Nullable`1 expectedOutputLength, String& output) in /_/src/Compilers/Test/Core/Platform/Desktop/RuntimeAssemblyManager.cs:line 390
   at System.Runtime.Remoting.Messaging.StackBuilderSink._PrivateProcessMessage(IntPtr md, Object[] args, Object server, Object[]& outArgs)
   at System.Runtime.Remoting.Messaging.StackBuilderSink.SyncProcessMessage(IMessage msg)

Exception rethrown at [0]: 
   at System.Runtime.Remoting.Proxies.RealProxy.HandleReturnMessage(IMessage reqMsg, IMessage retMsg)
   at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke(MessageData& msgData, Int32 type)
   at Roslyn.Test.Utilities.Desktop.RuntimeAssemblyManager.Execute(String moduleName, String[] mainArgs, Nullable`1 expectedOutputLength, String& output)
   at Roslyn.Test.Utilities.Desktop.DesktopRuntimeEnvironment.Execute(String moduleName, String[] args, String expectedOutput) in /_/src/Compilers/Test/Core/Platform/Desktop/DesktopRuntimeEnvironment.cs:line 244
   at Microsoft.CodeAnalysis.Test.Utilities.CompilationVerifier.Emit(String expectedOutput, Nullable`1 expectedReturnCode, String[] args, IEnumerable`1 manifestResources, EmitOptions emitOptions, Verification peVerify, SignatureDescription[] expectedSignatures) in /_/src/Compilers/Test/Core/CompilationVerifier.cs:line 214
   at Microsoft.CodeAnalysis.Test.Utilities.CommonTestBase.Emit(Compilation compilation, IEnumerable`1 dependencies, IEnumerable`1 manifestResources, SignatureDescription[] expectedSignatures, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, Action`1 assemblyValidator, Action`1 symbolValidator, EmitOptions emitOptions, Verification verify) in /_/src/Compilers/Test/Core/CommonTestBase.cs:line 153
   at Microsoft.CodeAnalysis.Test.Utilities.CommonTestBase.CompileAndVerifyCommon(Compilation compilation, IEnumerable`1 manifestResources, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 assemblyValidator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, EmitOptions emitOptions, Verification verify) in /_/src/Compilers/Test/Core/CommonTestBase.cs:line 67
   at Microsoft.CodeAnalysis.CSharp.Test.Utilities.CSharpTestBase.CompileAndVerify(Compilation compilation, IEnumerable`1 manifestResources, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 validator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, EmitOptions emitOptions, Verification verify) in /_/src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs:line 830
   at Microsoft.CodeAnalysis.CSharp.UnitTests.AttributeTests_CallerInfoAttributes.TestGoodCallerArgumentExpressionAttribute_SwapArguments() in /_/src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_CallerInfoAttributes.cs:line 101
----- Inner Stack Trace -----
   at System.Reflection.RuntimeAssembly.nLoadImage(Byte[] rawAssembly, Byte[] rawSymbolStore, Evidence evidence, StackCrawlMark& stackMark, Boolean fIntrospection, Boolean fSkipIntegrityCheck, SecurityContextSource securityContextSource)
   at System.Reflection.Assembly.Load(Byte[] rawAssembly)
   at Roslyn.Test.Utilities.Desktop.DesktopRuntimeUtil.LoadAsAssembly(String moduleName, ImmutableArray`1 rawAssembly, Boolean reflectionOnly) in /_/src/Compilers/Test/Core/Platform/Desktop/DesktopRuntimeUtil.cs:line 93

```

</details>

Reply:
> TestOptions.RegularPreview?

I'm thinking of introducing `Regular10` in `TestOptions` that will temporarily matches RegularPreview, but will change to use CSharp10 when released. This makes it simpler to switch those tests to use CSharp10 later. Does that make sense? If not, I'll stick to RegularPreview

Reply:
> I'm thinking of introducing Regular10 in TestOptions that will temporarily matches RegularPreview, but will change to use CSharp10 when released. This makes it simpler to switch those tests to use CSharp10 later. Does that make sense? If not, I'll stick to RegularPreview

I don't think that we would want the tests to be tied to C# 10 version in the long term.


---
In reply to: [602041555](https://github.com/dotnet/roslyn/pull/51952#discussion_r602041555) [](ancestors = 602041555)

Reply:
> The tests currently already skips verification, but still fails only on net472 test runs.

The failures look related to the TargetFramework used. There is a couple of things we can do:
1) Disable execution of compiled test scenarios unless unit-tests are run on CoreClr. For example, see conditional logic in CompileAndVerify calls in src\Compilers\CSharp\Test\Symbol\Symbols\DefaultInterfaceImplementationTests.cs.
2) Skip affected unit-tests (```[ConditionalFact(typeof(CoreClrOnly))]```) unless they are run on CoreClr. 


---
In reply to: [602039235](https://github.com/dotnet/roslyn/pull/51952#discussion_r602039235) [](ancestors = 602039235)

Reply:
Added ConditionalFact and it seems to work. The CI is expected to be green :tada: